pkgs/applications/networking/instant-messengers/zoom-us/default.nix

```diff
@@ -95,9 +95,28 @@ in mkDerivation {
         cp $icon $out/share/icons/hicolor/$path/apps/us.zoom.Zoom.png
     done
 
-    ln -s $out/share/zoom-us/zoom $out/bin/zoom-us
+
+    # Do the manual Qt wrapping, because existing hook doesn't support our usecase
+    qtWrapperArgs+=(--prefix XDG_DATA_DIRS : "$out/share/zoom-us")
+
+    find "$out/share/zoom-us" ! -type d -executable -print0 | while IFS= read -r -d "" file; do
+      patchelf --print-interpreter "$file" >/dev/null 2>&1 || continue
+      wrapQtApp "$file"
+    done
+
+    # it has to pass out var into wrapper
+    cat > $out/bin/zoom-us <<EOF
+    #!${bash}/bin/bash
+    (
+      cd $out/share/zoom-us
+      exec ./ZoomLauncher
+    )
+    EOF
+    chmod +x $out/bin/zoom-us
```

Comment:
this is exactly what makeWrapper does. I don't see any reason why you shouldn't first cd to zoom directory with all the produced wrappers (that being `zoom` `zopen` and `ZoomLauncher`).

```
qtWrapperArgs = [
  ...
 ''--run "cd ${placeholder "out"}/share/zoom-us"''
];

Reply:
comment also why that is needed.